server/services/teamsSummaryCountService.ts

```diff
@@ -218,4 +218,62 @@ export default class TeamsSummaryCountService {
       return { critical: 0, high: 0 }
     }
   }
+
+  /**
+   * Helper: Get Veracode VERY_HIGH, HIGH, MEDIUM, LOW vuln counts for a team's products
+   */
+  async getTeamVeracodeVulnerabilityCounts(
+    products: StrapiProduct[],
+    serviceCatalogueService: ServiceCatalogueService,
+  ): Promise<{ veryHigh: number; high: number; medium: number; low: number }> {
+    if (!Array.isArray(products) || products.length === 0) {
+      return { veryHigh: 0, high: 0, medium: 0, low: 0 }
+    }
+    try {
+      const allComponents = await serviceCatalogueService.getComponents()
+      const productIds = new Set(products.map(p => p.id))
+      const validComponents = allComponents.filter(component => {
+        const productId = component?.attributes?.product?.data?.id
+        return productId && productIds.has(productId)
+      })
+
+      const counts = validComponents.reduce(
+        (accumulator, component) => {
+          const summary = component.attributes?.veracode_results_summary as VeracodeResultsSummary | undefined
+          if (!summary?.severity) return accumulator
+
+          summary.severity.forEach(severity => {
+            severity.category.forEach(category => {
```

Comment:
Just thinking, to save all this you could have a severity type (enum, whatever) and check the severity is in that, then just use the actual severity as the key, then you can do `accumulator[category.severity]` maybe? Then the return object will always be an array of actual severity keys and their values?

Meant to say last time, could just use `total` instead of `accumulator`, easier to write 😄